test(oxfmt): Fix failing tests on Windows

leaysgur opened the pull request:
Introduce https://docs.github.com/en/get-started/git-basics/configuring-git-to-handle-line-endings

Without this, the `oxfmt` tests will fail because:

- The committed test fixtures use `<LF>` as the EOL character on MacOS
  - There are no formatting diffs
- On Windows CI, these EOL characters are automatically converted to `<CRLF>`
- The formatter then changes these back to `<LF>`
- This difference causes the CI to fail

Initially, I tried to set it globally, but since CI failed, limited it to the minimum scope... 🫤 

> https://github.com/oxc-project/oxc/pull/13803

leaysgur commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13801?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#13801** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13801?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13801?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#13797** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13797?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#13794** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13794?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=13801).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Dunqing commented:
### Merge activity

* **Sep 16, 7:28 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

leaysgur committed:
test(oxfmt): Fix failing tests on Windows (#13801)

Introduce https://docs.github.com/en/get-started/git-basics/configuring-git-to-handle-line-endings

Without this, the `oxfmt` tests will fail because:

- The committed test fixtures use `<LF>` as the EOL character on MacOS
  - There are no formatting diffs
- On Windows CI, these EOL characters are automatically converted to `<CRLF>`
- The formatter then changes these back to `<LF>`
- This difference causes the CI to fail

Initially, I tried to set it globally, but since CI failed, limited it to the minimum scope... 🫤

> https://github.com/oxc-project/oxc/pull/13803
